prost-types/src/datetime.rs

```diff
@@ -420,8 +422,8 @@ pub(crate) fn year_to_seconds(year: i64) -> (i128, bool) {
     let is_leap;
     let year = year - 1900;
 
-    // Fast path for years 1900 - 2038.
-    if year as u64 <= 138 {
+    // Fast path for years 1901 - 2038.
+    if (1..=138).contains(&year) {
```

Comment:
So basically, you are saying that the fast path was incorrect for 1900, but the slow path is correct. I can't reason why that is, but I assume that you are right :-)

Reply:
The [cited function in musl](https://git.musl-libc.org/cgit/musl/tree/src/time/__year_to_secs.c) actually writes this condition as `if (year-2ULL <= 136) {`... which isn't the same thing as `if year as u64 <= 138 {` at all! if year is either 0 _or_ 1 in the original function it wraps to max.

the musl code's fast-path works within i32 seconds of the epoch, which ranges from 1901-12-13 to 2038-01-18, so it actually skips the fast-path for 1901 as well because the start of that year is an underflow. for our code, we are calculating and returning an i128 anyway so it doesn't matter. (the code in this pr is still equally correct here if we write `if (1..=199).contains(&year) {` :) )

what DOES matter is that the fast-path ignores the century rule for leap years, because 2000 is a multiple of 400 and happens to be one! so the fast-path logic is actually WRONG when applied to any year outside 1901..=2099, even without integer wrapping.

Reply:
I understand how this could happen. That is a subtle, but important difference. Thanks for the extensive explanation!